be/src/vec/data_types/data_type_decimal.h

```diff
@@ -655,6 +656,10 @@ void convert_to_decimal(typename ToDataType::FieldType* dst,
         for (size_t i = 0; i < size; ++i) {
             dst[i].value = FromFieldType(src[i] * multiplier.value + ((src[i] >= 0) ? 0.5 : -0.5));
         }
+    } else if constexpr (IsDateV2Type<FromDataType>) {
+        for (size_t i = 0; i < size; ++i) {
+            dst[i].value = reinterpret_cast<const DateV2Value<DateV2ValueType>&>(src[i]).to_int64();
```

Comment:
1. maybe `assert_cast<const DateV2Value<FromFieldType>&>` is better
2. why convert result to int64? what if target Decimal is Decimal32 or Decimal128?